service/src/test/java/bio/terra/workspace/service/resource/controlled/ControlledResourceServiceTest.java

```diff
@@ -1957,6 +1958,92 @@ public void updateControlledBigQueryDatasetLifetime_onlyUpdateWhenLifetimesAreEm
         DEFAULT_CREATED_BIG_QUERY_PARTITION_LIFETIME);
   }
 
+  // Ensure the flight doesn't crash when there are BigQuery datasets with one null lifetime on the
+  // cloud.
+  @Test
+  public void updateControlledBigQueryDataset_HandlesDatasets_WithOneNullLifetime()
+      throws Exception {
+    ControlledBigQueryDatasetResource createdDataset =
+        makeBigQueryDatasetWithLifetime(
+            /*defaultTableLifetime=*/ null, DEFAULT_CREATED_BIG_QUERY_PARTITION_LIFETIME);
+
+    // Check which BQ datasets' lifetime to update.
+    List<ControlledBigQueryDatasetResource> emptyList =
+        updateControlledBigQueryDatasetsLifetimeAndWait();
+
+    // Update nothing because one of the lifetimes is populated (the SQL query checks for AND - both
+    // lifetimes null).
+    assertTrue(emptyList.isEmpty());
+
+    // Artificially set lifetimes to null in the database.
+    resourceDao.updateBigQueryDatasetDefaultTableAndPartitionLifetime(createdDataset, null, null);
+
+    List<ControlledBigQueryDatasetResource> updatedResourceList =
+        updateControlledBigQueryDatasetsLifetimeAndWait();
+
+    // The controlled dataset is updated since the lifetime is null.
+    assertEquals(1, updatedResourceList.size());
+    assertControlledBigQueryDatasetLifetimeIsUpdatedAndActivityIsLogged(
+        updatedResourceList,
+        createdDataset.getResourceId(),
+        /*expectedTableLifetime=*/ null,
+        DEFAULT_CREATED_BIG_QUERY_PARTITION_LIFETIME);
+  }
+  // Ensure the flight doesn't crash when there are BigQuery datasets with two null lifetimes on the
+  // cloud.
+  @Test
+  public void updateControlledBigQueryDataset_HandlesDatasets_WithTwoNullLifetimes()
```

Comment:
nit, can we just keep this test and not the one above? this seems to cover the one null case too.